src/measure_theory/function/strongly_measurable.lean

```diff
@@ -786,6 +786,92 @@ begin
   exact measurable_set_le hf.measurable.subtype_mk hg.measurable.subtype_mk,
 end
 
+lemma strongly_measurable_in_set {m : measurable_space α} [topological_space β] [has_zero β]
+  {s : set α} {f : α → β}
+  (hs : measurable_set s) (hf : strongly_measurable f) (hf_zero : ∀ x ∉ s, f x = 0) :
+  ∃ fs : ℕ → α →ₛ β, (∀ x, tendsto (λ n, fs n x) at_top (𝓝 (f x))) ∧ (∀ (x ∉ s) n, fs n x = 0) :=
+begin
+  let g_seq_s : ℕ → @simple_func α m β := λ n,
+  { to_fun := s.indicator (hf.approx n),
```

Comment:
Can you use `simple_func.restrict` here?

Reply:
Thanks, this is exactly the construction I needed and it made the proof much shorter.